File: library/src/main/java/com/qingmei2/rximagepicker/di/scheduler/RxImagePickerSchedulers.java
package com.qingmei2.rximagepicker.di.scheduler;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * An extra layer of packaging, using {@link IRxImagePickerSchedulers} for control thread switch.
 * <p>
 * Use {@link Schedulers#io()} and {@link AndroidSchedulers#mainThread()} directly is not
 * conducive to unit testing, you should use {@link RxImagePickerTestSchedulers} for inject {@link Scheduler}
 * in the testing.
 * <p>
 * Created by QingMei on 2018/1/18.
 */
public class RxImagePickerSchedulers implements IRxImagePickerSchedulers {

    @Override
    public Scheduler ui() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }
}

File: rximagepicker/src/main/java/com/qingmei2/rximagepicker/scheduler/RxImagePickerSchedulers.kt
package com.qingmei2.rximagepicker.scheduler

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * An extra layer of packaging, using [IRxImagePickerSchedulers] for control thread switch.
 *
 *
 * Use [Schedulers.io] and [AndroidSchedulers.mainThread] directly is not
 * conducive to unit testing, you should use [RxImagePickerTestSchedulers] for inject [Scheduler]
 * in the testing.
 */
class RxImagePickerSchedulers : IRxImagePickerSchedulers {

    override fun ui(): Scheduler {
        return AndroidSchedulers.mainThread()
    }

    override fun io(): Scheduler {
        return Schedulers.io()
    }
}
